### ISSUE TITLE
InFamous: Enemies take no damage and disappear/teleport

### ISSUE THREAD

**Issue by Dormant-Hero** [RPCS3/rpcs3#11719]
## Details

Enemies take no damage from my awesome attacks. They then disappear and still cannot be damaged until they reappear fully or do specific animations such as Molotov cocktail one not present in my evidence. 

#### 1. Please attach RPCS3's log.

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/8403597/RPCS3.log.gz)

#### 2. If you describe graphical issue, please provide an RSX capture and a RenderDoc capture that demonstrate it.

RSX Captures: https://1fichier.com/?73g9xzs7fzrwxwgnqg27
Would have done renderdoc but it refuses to hook on rpcs3 whilst running the game. 

#### 3. Please attach screenshots of your problem.

Before enemy disappears 
![Screenshot_2022 04 03 01 25 26 - YouTube — Mozilla Firefox_1](https://user-images.githubusercontent.com/79374258/161406664-d5ba83e2-17fb-4eee-ba91-bd3774a95ba8.png)

After enemy disappears
![image](https://user-images.githubusercontent.com/79374258/161406786-44588ebc-b928-4e28-8ef5-69bb50be7544.png)

Short video of the event https://youtu.be/fyov0Qq2dO8

(I realised after I could use F12 to capture images using rpcs3 lol).

#### 4. Please provide comparison with real PS3.

Too much effort, however very clear this is not supposed to happen anyway.

#### 5. Please provide your system configuration:
* OS Garuda
* Kernal 5.17.1-zen1-1-zen
* CPU i7 12700k
* GPU msi trio 3080ti
* Driver version Nvidia DKMS 510.54

**Comment by illusion0001** [comment]
sounds more like ppu/spu issue

**Comment by Dormant-Hero** [comment]
I am sure you are correct. Someone reported it in the past in a forum (https://www.reddit.com/r/rpcs3/comments/l3pl4s/couple_questions_about_enemies_and_firing_in/) and I know Yuri recognised this and thought the issue was reported. On further inspection could not find on github.

**Comment by illusion0001** [comment]
try interpreters

**Comment by Dormant-Hero** [comment]
I am on windows now with the same save and the exact same thing happens, so that at least takes OS out of the equation. 

I now tried interpreters for both the the below. Hopefully this is what you meant. It did not solve the issue and the same thing happens for me. 

![image](https://user-images.githubusercontent.com/79374258/161407299-3e0049e7-089f-4e0d-a05a-80a03dd00ab2.png)

**Comment by Darkhost1999** [comment]
This is the last known improvement to this issue
https://github.com/RPCS3/rpcs3/pull/11151#issuecomment-968310176
It was solved for a good 2 to 3 hours of gameplay and has maintained consistency for being less likely to occur.
Essentially saying before this PR the issue is far more common.

**Comment by Xcedf** [comment]
Dissapearing enemies in Infamous is SPU Xfloat precision issue
you can use Accurate XFloat in this game or hacky versions of emulator to improve precision of certain Xfloat SPU instructions.

**Comment by Yahfz** [comment]
> Dissapearing enemies in Infamous is SPU Xfloat precision issue you can use Accurate XFloat in this game or hacky versions of emulator to improve precision of certain Xfloat SPU instructions.

Which SPU instruction has to be accurate to make it work properly?

**Comment by Xcedf** [comment]
@Yahfz I mostly use hacked version and enable accuracy hacks for FNMS, FMA and FM, perhaps not all of them are required

**Comment by Dormant-Hero** [comment]
When I get a couple of hours to test then I will try enabling accuracy options in the emu if anyone needs me to. Just make it clear what I need to do or pass me a custom config file for infamous and I will try it. Current config used below, but worked fine for a smooth experience, so did not have to edit much. I guess things will be getting more accurate if it is a simple matter of changing the config. 

You guys probably have infamous anyway I guess. 

[config_BCES00609.zip](https://github.com/RPCS3/rpcs3/files/8404370/config_BCES00609.zip)

**Comment by Dormant-Hero** [comment]
HI All,

I tested accurate xfloat twice just now in that one scene loading the exact same save that would reproduce the same issues every time I tried without accurate xfloat.

I can confirm it works. It is a game breaking issue with more enemies on screen, therefore I think it would be great if discord bot would pick it up from logs as well as it to be listed on the below site. Not sure if this happens for everyone though.

https://wiki.rpcs3.net/index.php?title=InFamous
![image](https://user-images.githubusercontent.com/79374258/161452007-e2605920-64e8-4256-ae23-e31aedb61a2c.png)

**Comment by Dormant-Hero** [comment]
I managed to complete the game with xfloat approx and accurate ticked which removed the issues mentioned initially.

People mentioned that there would be an issue defeating the last boss, but I experienced nothing game breaking here.

The only issues I had with the game once xfloat accurate used are slight graphical issues. The electric shield, electric float/propulsion, fire, and anything that produced air/wind are all graphically like a mirror. It is hard to explain.

Anyway I will close this issue as just a setting change to fix the enemy disappearance act and damage issue.
